plugins/woocommerce/includes/class-wc-product-simple.php

```diff
@@ -74,4 +74,27 @@ public function add_to_cart_description() {
 
 		return apply_filters( 'woocommerce_product_add_to_cart_description', sprintf( $text, $this->get_name() ), $this );
 	}
+
+	/**
+	 * Get the add to cart button success message - used to update the mini cart live region.
+	 *
+	 * @return string
+	 */
+	public function add_to_cart_success_message() {
+		/* translators: %s: Product title */
+		$text = $this->is_purchasable() && $this->is_in_stock() ? __( '&ldquo;%s&rdquo; has been added to your cart', 'woocommerce' ) : '';
+
+		if ( ! empty( $text ) ) {
+			$text = sprintf( $text, $this->get_name() );
+		}
```

Comment:
We can probably avoid the `empty()` check and make this more readable by encapsulating with the ternary condition:

```suggestion
        $text = '';
        if ( $this->is_purchasable() && $this->is_in_stock() ) {
            /* translators: %s: Product title */
            $text = __( '&ldquo;%s&rdquo; has been added to your cart', 'woocommerce' );
            $text = sprintf( $text, $this->get_name() );
        }
```

Reply:
@wavvves Please review it: [Refactor logic in the add_to_cart_success_message function](https://github.com/woocommerce/woocommerce/pull/48295/commits/728af717e0dc75479b50e9d88b2756f9d3647171)